sagemathinc/cocalc
# Typescript project files

# Description
Move ProjectFiles into its own directory under `/project` and rename to `Explorer`.
Split the many components into their own files.
Rewrite some components to have smaller scope.

# Testing Steps
Compare behavior to Production:
1. Open a project
1. Create a file using the +New button
1. Create a file using the search box
1. Copy a file
1. Duplicate a file
1. Rename a file
1. Move a file
1. Share a file
1. Delete a file
1. Download file
1. Make enough file to span multiple pages
1. Click next and prev page buttons
1. Click "Check all" button
1. Click "Select all __ items" button
1. Click clear selection
1. Use the mini-term
1. Click the misc buttons in the right menu bar (like the visibility eyeball)
1. Sort files by column names
1. Navigate with file path breadcrumbs

# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):

Front end:
